### DISCUSSION TITLE
operator that store the id returned by an external API

### DISCUSSION THREAD

**Discussion by briner** [nolar/kopf#1078] [Q&A]
I we would like to create an operator that works with an external API. This API will be slow (about 1h) to realize the change. For earch CR (instance of the new CRD), we will make a call. The API will reply with an objectID, that we could use to poll the status. How can we store this objectID (in a k8s way). And which pattern should we use to follow a such long process.

Do you have any example on how to do this ?

Thanks in advance for your help !

**Comment by bobh66** [comment]
The kubernetes way is to create a status field in your CR for the remote ID and store it there.  Then use that field value to poll the remote state.

**Comment by nolar** [comment]
👆 What @bobh66 said above, plus Kopf's [daemons](https://kopf.readthedocs.io/en/stable/daemons/) and/or [timers](https://kopf.readthedocs.io/en/stable/timers/) to poll the status of the remote entity and do something when it changes.

**Comment by briner** [comment]
Thank you all of you (@bobh66, @nolar) for your quick responses.

Last question as I do not know all the internal of kubernetes : Is it safe to store the data on the status of the CR. As this part seems to be editable with kubectl

**Comment by bobh66** [comment]
It's as safe as any other field - spec and metadata are also editable via kubectl
